# rs_user_clean

In [34]:
%matplotlib inline
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
import io
import os

# rs_user

### usuarios navegando

In [35]:
data_user="../DATA/IntropiaCSV/rs_user.csv"

readme=os.path.join(data_user)
#cat $readme | head

In [36]:
datafile = os.path.join(data_user)
data_user = pd.read_csv(datafile, sep=';')
data_user.head(0)


Empty DataFrame
Columns: [iduser, tags, geolocation, email, description, name, active, imageurl, url, misc, timestamp]
Index: []

In [37]:
data_user.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48969 entries, 0 to 48968
Data columns (total 11 columns):
iduser         48969 non-null int64
tags           0 non-null float64
geolocation    0 non-null float64
email          48969 non-null object
description    0 non-null float64
name           0 non-null float64
active         48969 non-null int64
imageurl       0 non-null float64
url            0 non-null float64
misc           48968 non-null object
timestamp      48969 non-null int64
dtypes: float64(6), int64(3), object(2)
memory usage: 4.1+ MB


In [38]:
# Hay muchas coluans sin valores. No nos aportan nada.

In [39]:
# seleccionamos columnas interés.

user = data_user[['iduser','active','misc','timestamp']]

# vistazo

user.head()

iduser  active   misc   timestamp
0  1000004887844       1     es  1459288467
1  2000000000333       1  es_es  1376468347
2  2000000000334       0  es_es  1376464186
3  2000000000335       1     es  1485442621
4  2000000000336       1  gb_en  1376484799

Vamos a ir a la carpeta donde estan los datos originales, y a la tabla rs_user le cambiamos el nombre y la renombramos con el nombre rs_user_sec.Esta tabla tiene datos sensibles, que serán eliminados. Por lo tanto no se sube al repositorio.
Y esta con las columnas que nos interesan las guardamos con el nombre actual. Lo hacemos así ya que tenemos mucho ya con este nombre, y puede dar a error.

In [40]:
# guardamos la tabla: 

#Save into a csv

user.to_csv('../DATA/IntropiaCSV/rs_user_prueba.csv')

In [41]:
# Calculate number of diferents elements >> unicos

n_user =data_user.iduser.unique().shape[0]
n_email = data_user.email.unique().shape[0]
n_active = data_user.active.unique().shape[0]
n_misc = data_user.misc.unique().shape[0]

print("There are %s users" % (n_user))
print("There are %s email" % (n_email))
print("There are %s active" % (n_active))
print("There are %s misc" % (n_misc))

There are 48969 users
There are 48717 email
There are 2 active
There are 171 misc


Notas: Tenemos menos emails que usuarios, esto puede deberse a que son guest o bien tienen cuenta.
En cualquier caso, no nos importa, ya que lo relevante es el "id".
Tenemos una combinación de 171 misc. Esta es una combinación de país-idioma por site. 
Hay web site que tienen 2 idiomas y otros que tienen 1.
Puede ser relevante a la hora de recomendar, dados loa hábitos de compra, climatología...pero inicialemente
no lo contemplo.
    


In [31]:
# otra forma de hacerlo, mediante una función q retorne 

def count_unique(column):
    return len(column.unique())

data_user[['iduser','email','active','misc']].apply(count_unique, axis=0).astype(np.int32)

iduser    48969
email     48717
active        2
misc        171
dtype: int32

In [45]:
# Vistazo a la cantidad de emails que son recurrentes.

# email_count = user.groupby('email').size().sort_values(ascending=False)
# print (email_count)

# Esta columna tiene relevancia solo para uso interno:
# Reflexión para dto. de marketing:
# Tenemos muy pocos emails recurrentes, esto se puede traducir en q tenemos muy pocos usuarios recurrentes???
# La bbbdd de emails es correcta >> revisar de forma conjunta

In [46]:
# Vistazo rápido activo/no activo.>> deduzco q es un indicador de un estado interno.

active_count = user.groupby('active').size().sort_values(ascending=False)
print (active_count)

active
1    41873
0     7096
dtype: int64


Sobre la columna "misc" >> que significan las útimas URL's??? se trata de enlaces rotos:

Posibilidades:

- vienen de emails recuperación carritos? 
- se deben a la incidencia que hubo en la página respecto al redireccionamiento? >> comprobar.

En cualquier caso, inicialmente no se tiene en cuenta para ningún calculo importante.

In [48]:
misc = user['misc'].value_counts()
misc


es                                                                                                     33048
es_es                                                                                                   5086
gb                                                                                                      2584
pt                                                                                                      1286
it                                                                                                      1075
gb_en                                                                                                    859
fr                                                                                                       795
de                                                                                                       787
it_en                                                                                                    405
pt_en              

In [ ]:
# Explorando

In [49]:
result = user['active'] == 0
len(result)

48969

In [50]:
len(user[result])

7096

In [51]:
user[result].head()

iduser  active   misc   timestamp
2   2000000000334       0  es_es  1376464186
8   2000000000340       0  be_en  1378466593
10  2000000000342       0  pt_en  1421016563
11  2000000000343       0  it_en  1376468432
14  2000000000346       0  es_es  1412971285

In [52]:
# Cositas varias probando.

In [56]:
len(user)
user.shape

(48969, 4)

In [58]:
user.count()

iduser       48969
active       48969
misc         48968
timestamp    48969
dtype: int64